src/NeoSharp.Core/Messaging/Handlers/GetAddrMessageMessageHandler.cs

```diff
@@ -39,11 +39,11 @@ public GetAddrMessageMessageHandler(IServer server)
 
             var networkAddressWithTimes = peers
                 .Select(p => new NetworkAddressWithTime
-                    {
-                        EndPoint = p.EndPoint,
-                        Services = p.Version.Services,
-                        Timestamp = p.Version.Timestamp
-                    }
+                {
+                    EndPoint = p.EndPoint.ToEndPoint(),
```

Comment:
Why do we need this conversion?

Reply:
Neo protocol only receive and send IpEndPoint, in a custom format

Reply:
If we have changed the protocol already could we change this as well?

Reply:
But the old protocol need a endPoint, maybe the handler need to receive the protocol version

Reply:
You are right. We half implemented protocol switching. To make it right we need to support not only different serialization format but also message and communication order changes. We will leave this for now.